torch/headeronly/util/shim_utils.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <torch/headeronly/macros/Macros.h>
+
+#include <sstream>
+#include <stdexcept>
+
+#define TORCH_SUCCESS 0
+#define TORCH_FAILURE 1
+
+namespace torch::headeronly::detail {
+C10_NOINLINE inline void throw_exception(
```

Comment:
Sorry for the churn, but looking at this conflicting declaration `C10_NOINLINE inline`, I realized my previous suggestion doesn't make sense. The reason of having `C10_NOINLINE` here is because it was causing very slow compilation. I think we still need the old way of declaring this as `static`.